packages/ses/src/module-instance.js

```diff
@@ -4,6 +4,77 @@ import { create, entries, keys, freeze, defineProperty } from './commons.js';
 // q, for enquoting strings in error messages.
 const q = JSON.stringify;
 
+export function makeThirdPartyModuleInstance(
+  compartmentPrivateFields,
+  staticModuleRecord,
+  compartment,
+  moduleAliases,
+  moduleSpecifier,
+  resolvedImports,
+) {
+  const { exportsProxy, proxiedExports, activate } = getDeferredExports(
+    compartment,
+    compartmentPrivateFields.get(compartment),
+    moduleAliases,
+    moduleSpecifier,
+  );
+
+  const notifiers = create(null);
+
+  if (staticModuleRecord.exports) {
+    if (
+      !Array.isArray(staticModuleRecord.exports) ||
+      staticModuleRecord.exports.some(name => typeof name !== 'string')
+    ) {
+      throw new TypeError(
+        `SES third-party static module record "exports" property must be an array of strings for module ${moduleSpecifier}`,
+      );
+    }
+    staticModuleRecord.exports.forEach(name => {
+      let value = proxiedExports[name];
+      const updaters = [];
+
+      const get = () => value;
+
+      const set = newValue => {
+        value = newValue;
+        for (const updater of updaters) {
+          updater(newValue);
+        }
+      };
+
+      defineProperty(proxiedExports, name, {
+        get,
+        set,
+        enumerable: true,
+        configurable: false,
+      });
+
+      notifiers[name] = update => {
+        updaters.push(update);
+        update(value);
+      };
+    });
+  }
+
+  let activated = false;
+  return freeze({
+    notifiers,
+    exportsProxy,
+    execute() {
+      if (!activated) {
+        activate();
+        activated = true;
+        staticModuleRecord.execute(
+          proxiedExports,
+          compartment,
+          resolvedImports,
```

Comment:
This argument order is a bit odd. Is there a reason you chose this order?

Reply:
Arbitrary but not random, and subject to change. I strived to move from generic to specific in the argument order.